sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/FunctionRegistry.scala

```diff
@@ -407,6 +407,7 @@ object FunctionRegistry {
     expression[MapKeys]("map_keys"),
     expression[MapValues]("map_values"),
     expression[Size]("size"),
+    expression[Cardinality]("cardinality"),
```

Comment:
`expression[Size]("cardinality"),` Does this work?

Reply:
In Presto, `cardinality`'s return type is `BigInt`. Thus, `cardinality` in Spark uses uses `long` as return type. If we use `int` as `cardinality`'s return type, I think that it works.

Reply:
ping @gatorsmile

Reply:
That might be fine in most cases. If we really need it In the future, we can add new APIs with better names later?

Like in SQL, we have COUNT and COUNT_BIG. COUNT_BIG is similar to COUNT but the result can be grater than the max value of integer.

Reply:
I see. I will update this PR with only `expression[Cardinality]("cardinality")`. I will also update the description and JIRA later.

Is it OK with you? @gatorsmile cc: @ueshin

Reply:
One question: Do we need `@ExpressionDescription`? If we need this, we may need to create `Cardinality` case class that is the same as `Size`. Of course, `Cardinality` and `Size` can extend the same parent class.

Reply:
Alias names are implemented like that. No need to create a new one. We did it in the other SQL functions too. like char,  char_length and many. We can update the description in the existing ExpressionDescription

Reply:
Thank you for your clarification. I see.